File: 프로그래머스/lv2/42586. 기능개발/기능개발.py
def solution(progresses, speeds):
    answer = []
    
    while len(progresses)>0:
        done=0
      
        for i in range(len(progresses)):
            progresses[i]=progresses[i]+speeds[i]

              
        for j in range(len(progresses)):
            if progresses[0]>=100:
                done=done+1
                del progresses[0]
                del speeds[0]
            else:
                break
        
        
        if done>0:
            answer.append(done)
        
    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        int n = progresses.length;
        int[] leftDays = new int[n];
        for(int i = 0; i<n; i++){
            leftDays[i] = (int)Math.ceil((100-progresses[i])/(double)speeds[i]);
        }
        ArrayList<Integer> answer = new ArrayList<>();
        ArrayDeque<Integer> queue = new ArrayDeque<>();
        for(int left: leftDays) queue.addLast(left);
        
        int cnt=1;
        int temp = queue.pollFirst();
        while(!queue.isEmpty()){
            if(temp>=queue.peekFirst()){
                queue.pollFirst();
                cnt+=1;
            }
            else{
                temp = queue.peekFirst();
                answer.add(cnt);
                cnt = 0;
            }
        }
        answer.add(cnt);
        
        return answer.stream().mapToInt(Integer::intValue).toArray();
    }
}